ImageFinder/View/SearchImageViewController.swift

```diff
@@ -108,44 +159,83 @@ extension SearchImageViewController: UICollectionViewDataSource {
     print(indexPath.row)
     let urlString = resultImages[indexPath.row].thumbnailUrl
 
-    if let url = URL(string: urlString) {
-      if let data = try? Data(contentsOf: url) {
-        let image = UIImage(data: data)
-        print(data)
-        cell.thumbnailView.image = image
-        print(urlString)
+    DispatchQueue.global().async {
+      if let url = URL(string: urlString) {
+        if let data = try? Data(contentsOf: url) {
+          let image = UIImage(data: data)
+          DispatchQueue.main.async {
+            cell.thumbnailView.image = image
+          }
+        }
       }
     }
-//    cell.thumbnailView.image = UIImage(named: "jjong")
     return cell
   }
 }
 
 extension SearchImageViewController: UISearchBarDelegate {
-  func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
-    print("heyhey")
-    let time = DispatchTime.now() + .seconds(1)
-    let keyword = searchBar.text ?? ""
-
-    resultCollectionView.isHidden = true
-    searchLodingIndicator.startAnimating()
-
-    DispatchQueue.main.asyncAfter(deadline: time) {
-      self.searchLodingIndicator.stopAnimating()
-      self.resultCollectionView.isHidden = false
-      self.presenter.imageSearchService.doSearchImage(keyword: keyword)
-      self.resultCollectionView.reloadData()
+  func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
+    setStatusView(status: .keywordTyping)
+  }
+
+  func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+    searchDelayer.invalidate()
+    self.searchLoadingIndicator.startAnimating()
+    resultImages = []
+    page = 1
+
+    if true {
+      searchDelayer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(self.doDelayedSearch), userInfo: searchText, repeats: false)
+    }
+  }
+
+  @objc func doDelayedSearch(_ timer: Timer) {
+    if let keyword = searchDelayer.userInfo as? String {
+      searchLoadingIndicator.stopAnimating()
+      resultCollectionView.isHidden = false
+      presenter?.setResultImage(keyword: keyword, page: self.page)
     }
+    searchDelayer.invalidate()
   }
 }
 
 extension SearchImageViewController: UICollectionViewDelegate {
   func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
     let imageDetailView = ImageDetailViewController()
 
-    imageDetailView.imageUrl = "jjong"
+    let urlString = resultImages[indexPath.row].imageUrl
+    let rowDateTime = resultImages[indexPath.row].dateTime
+    let imageSource = resultImages[indexPath.row].displaySiteName
+
+    if let url = URL(string: urlString) {
+      if let data = try? Data(contentsOf: url) {
+        let image = UIImage(data: data)
+        print(data)
+        imageDetailView.imageView.image = image
+      }
+    }
```

Comment:
```suggestion
    if let url = URL(string: urlString),
       let data = try? Data(contentsOf: url) {
        let image = UIImage(data: data)
        print(data)
        imageDetailView.imageView.image = image
      }
```
이중으로 옵셔널 바인딩을 하는것보다 이게 보기에 좋을거 같습니다.